src/endpoint.jl

```diff
@@ -145,8 +145,8 @@ function Base.isless(a::RightEndpoint, b::LeftEndpoint)
 end
 
 # Comparisons between Scalars and Endpoints
-Base.:(==)(a, b::Endpoint) = a == b.endpoint && isclosed(b)
-Base.:(==)(a::Endpoint, b) = b == a
+Base.:(==)(a::Number, b::Endpoint) = a == b.endpoint && isclosed(b)
+Base.:(==)(a::Endpoint, b::Number) = b == a
```

Comment:
I'm pretty sure we can update this to be this safely (tests will show if that is the case):
```suggestion
Base.:(==)(a::T, b::Endpoint{T}) where T = a == b.endpoint && isclosed(b)
Base.:(==)(a::Endpoint{T}, b::T) where T = b == a
```
I'm not sure these are actually used so we may be able to just drop them.